kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveKeyValueStoreRestoreIntegrationTest.java

```diff
@@ -351,30 +350,33 @@ private RemoteKVTable<?> remoteKVTable(
   ) throws InterruptedException, TimeoutException {
     final RemoteKVTable<?> table;
 
-    if (type == KVSchema.FACT) {
-      final CassandraClient cassandraClient = defaultFactory.createClient(
-          defaultFactory.createCqlSession(config, null),
-          config);
+    final CassandraClient cassandraClient = defaultFactory.createClient(
+        defaultFactory.createCqlSession(config, null),
+        config);
 
+    if (type == KVSchema.FACT) {
       table = cassandraClient.factFactory()
           .create(new DefaultTableSpec(
               aggName(),
               TablePartitioner.defaultPartitioner(),
               TtlResolver.NO_TTL,
               config
           ));
-
     } else if (type == KVSchema.KEY_VALUE) {
-      final var connectionString = config.getPassword(MONGO_CONNECTION_STRING_CONFIG).value();
-      final var mongoClient = SessionUtil.connect(connectionString, "", null);
-      table = new MongoKVTable(
-          mongoClient,
+      final SubPartitioner partitioner = SubPartitioner.create(
+          OptionalInt.empty(),
+          NUM_PARTITIONS,
           aggName(),
-          CollectionCreationOptions.fromConfig(config),
-          TtlResolver.NO_TTL,
-          config
+          config,
+          changelog.topic()
       );
-      table.init(0);
+      table = cassandraClient.kvFactory()
```

Comment:
@ableegoldman I had to partially revert https://github.com/responsivedev/responsive-pub/pull/383 here. One of the tests `shouldRestoreUnflushedChangelog` used a cassandra fault injector which did nothing when MONGO was installed as the backend.